tests/system/aiplatform/test_e2e_vertex_model.py

```diff
@@ -0,0 +1,152 @@
+        feature_columns = ["feat_1", "feat_2"]
+        data = pd.DataFrame(instances, columns=feature_columns)
+        torch_tensor = torch.tensor(data[feature_columns].values).type(
+            torch.FloatTensor
+        )
+        return torch_tensor
+
+    # Implementation of predict_input_to_predict_payload(), which converts predict() inputs to a predict_payload object
+    def predict_input_to_predict_payload(self, parameter):
+        return parameter.tolist()
+
+    # Implementation of predict_output_to_predict_payload(), which converts the predict() output to a predict_payload object
+    def predict_output_to_predict_payload(self, output):
+        return output.tolist()
+
+    # Implementation of predict_payload_to_predict_output, which takes a predict_payload object containing predictions and
+    # converts it to the type of output expected by the user-written class.
+    def predict_payload_to_predict_output(self, predictions):
+        data = pd.DataFrame(predictions)
+        torch_tensor = torch.tensor(data.values).type(torch.FloatTensor)
+        return torch_tensor
+
+
+@pytest.mark.usefixtures("prepare_staging_bucket", "delete_staging_bucket", "teardown")
+class TestEndToEndVertexModel(e2e_base.TestEndToEnd):
+    """End to end system test of the Vertex SDK with the VertexModel interface"""
+
+    _temp_prefix = "ucaip-mb-sasha-dev"
+
+    @pytest.mark.parametrize("training_remote", [True, False])
+    @pytest.mark.parametrize("prediction_remote", [True, False])
+    def test_end_to_end_vertex_model(
+        self, shared_state, training_remote, prediction_remote
+    ):
+        # Collection of resources generated by this test, to be deleted during teardown
+        shared_state["resources"] = []
+
+        aiplatform.init(
+            project=e2e_base._PROJECT,
+            location=e2e_base._LOCATION,
+            staging_bucket="gs://" + shared_state["staging_bucket_name"],
+        )
+
+        # Set up random training data
+        df = pd.DataFrame(
+            np.random.random(size=(100, 3)), columns=["feat_1", "feat_2", "target"]
+        )
+
+        # Train using VertexModel interface
+        my_model = LinearRegression(2, 1)
+        my_model.remote = training_remote
+
+        my_model.fit(df, "target", 1, 0.1)
+
+        if training_remote:
+            shared_state["resources"].extend([my_model._training_job])
+            shared_state["resources"].extend([my_model._model])
+
+        # Set up test data
+        data = pd.DataFrame(
+            np.random.random(size=(100, 3)), columns=["feat_1", "feat_2", "target"]
+        )
+
+        feature_columns = list(data.columns)
+        feature_columns.remove("target")
+
+        torch_tensor = torch.tensor(data[feature_columns].values).type(
+            torch.FloatTensor
+        )
+
+        # Prediction with trained model
+        my_model.remote = prediction_remote
+        predictions = my_model.predict(torch_tensor)
+
+        if prediction_remote:
```

Comment:
In the local train to remote prediction CUJ do we have a handle to the uploaded model we can add as well?

Reply:
yes, because when the model is uploaded we assign it to the obj._model variable.